app/ide-desktop/lib/dashboard/src/services/RemoteBackend.ts

```diff
@@ -28,14 +28,19 @@ const STATUS_SUCCESS_LAST = 299
 const STATUS_NOT_FOUND = 404
 /** HTTP status indicating that the server encountered a fatal exception. */
 const STATUS_SERVER_ERROR = 500
-/**
- * HTTP status indicating that the request was successful, but the user is not authorized to access
- */
+/** HTTP status indicating that the request was successful, but the user is not authorized to access. */
 const STATUS_NOT_AUTHORIZED = 401
 
 /** The number of milliseconds in one day. */
 const ONE_DAY_MS = 86_400_000
 
+/** The internal port of the JSON WebSocket server in a Cloud VM. */
+const JSON_WS_SERVER_PORT = 30001
+/** The `ls` parameter for the YDoc server in a Cloud VM. */
+const YDOC_QUERY_STRING = new URLSearchParams({
+  ls: `ws://localhost:${JSON_WS_SERVER_PORT}`,
```

Comment:
If it's cloud why localhost then?

Reply:
the ydoc server is on the same vm as the language server, so from the point of view of the ydoc server the language server is on localhost